hyperspy/drawing/utils.py

```diff
@@ -343,6 +343,466 @@ def _set_spectrum_xlabel(spectrum, ax):
+            If 'ticks', no axis labels will be shown, but ticks and tick labels will
+            If 'off', all decorations and frame will be disabled
+            If None, no axis decorations will be shown, but ticks/frame will
+
+        padding : None or dict
+            This parameter controls the spacing between images. If None, default options will be used
+            Otherwise, supply a dictionary with the spacing options as keywords and desired values as values
+            Values should be supplied as used in pyplot.subplots_adjust(), and can be:
+                'left', 'bottom', 'right', 'top', 'wspace' (width), and 'hspace' (height)
+
+        tight_layout : bool
+            If true, hyperspy will attempt to improve image placement in figure using matplotlib's tight_layout
+             This is known to cause some problems, so an option is provided to disable it.
+             Turn this option off if output is not as expected, or try adjusting `labelwrap` or `per_row`
+            If false, repositioning images inside the figure will be left as an exercise for the user.
+
+        aspect : str or (float, int, long)
+            If 'auto', aspect ratio will be automatically determined, subject to min_asp.
+            If 'square', image will be forced onto square display.
+            If 'equal', aspect ratio of 1 will be enforced.
+            If float (or int/long), given value will be used.
+
+        min_asp : float
+            Minimum aspect ratio to be used when plotting images
+
+        fig : mpl figure
+            If set, the images will be plotted to an existing MPL figure
+
+        *args, **kwargs
+            Additional arguments passed to matplotlib.imshow()
+
+        Returns
+        -------
+        axes_list, a list of subplot axes that hold the images
+
+    """
+    from mpl_toolkits.axes_grid1 import make_axes_locatable
+    from hyperspy.drawing import widgets
+    from hyperspy.misc import rgb_tools
+    from hyperspy.signal import Signal
+    from traits import trait_base
+    from itertools import compress
+    import textwrap
+
+    if isinstance(images, Signal) and len(images) is 1:
+        print "Single image provided, using Signal.plot() instead."
+        print "Use Signal.plot() directly to supply more options."
+        images.plot()
+        f = plt.gcf()
+        return f
+    elif not isinstance(images, (list, tuple, Signal)):
+        raise ValueError("images must be a list of image signals or multi-dimensional signal."
+                         " " + repr(type(images)) + " was given.")
+
+    # Get default colormap from pyplot:
+    if cmap is None:
+        cmap = plt.get_cmap()
+
+    # If input is >= 1D signal (e.g. for multi-dimensional plotting),
+    # copy it and put it in a list so labeling works out as (x,y) when plotting
+    if isinstance(images,Signal) and images.axes_manager.navigation_dimension > 0:
+        images = [images._deepcopy_with_new_data(images.data)]
+
```

Comment:
@pburdet, do you have a 3D RGB image that I can test with (or how to generate one)? I'd like to get it to work, if not too tricky.